core/integration/features/orders/stoporders.feature

```diff
@@ -0,0 +1,1146 @@
+    # place auxiliary orders so we always have best bid and best offer as to not trigger the liquidity auction
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     |
+      | aux   | ETH/DEC19 | buy  | 1      | 1     | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux   | ETH/DEC19 | sell | 1      | 10001 | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux2  | ETH/DEC19 | buy  | 5      | 50    | 0                | TYPE_LIMIT | TIF_GTC |
+      | aux   | ETH/DEC19 | sell | 5      | 50    | 0                | TYPE_LIMIT | TIF_GTC |
+
+    Then the opening auction period ends for market "ETH/DEC19"
+    And the trading mode should be "TRADING_MODE_CONTINUOUS" for the market "ETH/DEC19"
+
+    # setup party1 position, open a 10 short position
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     |
+      | party1| ETH/DEC19 | buy  | 10     | 50    | 0                | TYPE_LIMIT | TIF_GTC |
+      | party2| ETH/DEC19 | sell | 10     | 50    | 1                | TYPE_LIMIT | TIF_GTC |
+
+
+    # create party1 stop order, results in a trade
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     | only  | fb price trigger | ra price trigger | error | reference |
+      | party1| ETH/DEC19 | sell | 10     |  0    | 0                | TYPE_MARKET| TIF_IOC | reduce| 25               | 75               |       | stop1     |
+
+    # volume for the stop trade
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     |
+      | party3| ETH/DEC19 | buy  | 10     | 20    | 0                | TYPE_LIMIT | TIF_GTC |
+
+      # now we trade at 75, this will breach the trigger
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     |
+      | party3| ETH/DEC19 | sell | 10     | 25    | 0                | TYPE_LIMIT | TIF_GTC |
+      | party2| ETH/DEC19 | buy  | 10     | 25    | 1                | TYPE_LIMIT | TIF_GTC |
+
+    # check that the order got submitted
+    Then the orders should have the following states:
+      | party  | market id | side | volume | price | status        | reference |
+      | party1 | ETH/DEC19 | sell | 10     | 0     | STATUS_FILLED | stop1-1   |
+
+    Then the stop orders should have the following states
+      | party  | market id | status           | reference |
+      | party1 | ETH/DEC19 | STATUS_TRIGGERED | stop1-1   |
+      | party1 | ETH/DEC19 | STATUS_STOPPED   | stop1-2   |
+
+  Scenario: If a pair of stop orders are specified as OCO and one triggers but is invalid at time of triggering (e.g. a buy when the trader is already long) the other will still be cancelled. (0014-ORDT-058)
```

Comment:
I'm not sure this one is fully testing that the other side of an OCO is cancelled/stopped, looks like it just does a single stop order?

Reply:
nope it actually check if it's stopped, basically there's a fallsBellow and a risesAbove trigger in the stop order, when there's both it just happened a -1 and -2 to the reference (here stop1), that's how we find them. In this given case, the first one get triggered, the second oned stopped.